### PR TITLE
Cache-on-failure support

### LINKED ISSUE #20: Enable optional save cache on failure
In line with https://github.com/actions/cache/issues/92 but for here. Not sure how best to configure the `post-if` to have this as an option? Note I'm proposing this as a default-off option, not a change of defaults.

### PR DESCRIPTION
**Pull request by palfrey**
Fixes #20 

https://github.com/palfrey/rust-cache/pull/1 demos this running. https://github.com/palfrey/rust-cache/runs/2926086671 for example is an earlier run with the cache being done despite the job failing (which then fails itself in that case because not enough Rust bits had run to make it viable)

The `dist` bits have been regenerated with `npm run prepare` but I'm seeing some notable changes?

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -14,14 +14,17 @@ inputs:
   target-dir:
     description: "The target dir that should be cleaned and persisted, defaults to `./target`"
     required: false
+  cache-on-failure:
```
Comment (Swatinem) [inline action.yml:17]: can you put this into the readme as well?

**Review by Swatinem — APPROVED** [review]
lgtm!

I will run the build from my system after merging, so you might want to remove the dist file changes from the PR

File: action.yml (hunk 1 above)
Comment (palfrey) [reply action.yml:17]: Done

**Comment by palfrey** [comment]
> I will run the build from my system after merging, so you might want to remove the dist file changes from the PR

Done

**Pull request merged**